### PR TITLE
feat: add PWA features to support Add to Home Screen

### PR DESCRIPTION
**Pull request by ThaUnknown**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[X] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
[Hopefully] Added a working manifest and service worker to make the website installable as a PWA. This is a bare minimum, the service worker has no added functionality like caching etc.
**Which issue (if any) does this pull request address?**
https://github.com/webtorrent/instant.io/issues/213

### CODE REVIEW

**Review by hello-smile6 — APPROVED** [review]
Good! Just add the `apple-mobile-web-app-capable` meta tag, and this is ready to go!

**Comment by hello-smile6** [comment]
You'll need a separate icon for this. iOS does black for transparency.

File: static/sw.js [hunk 1]
```diff
@@ -0,0 +1 @@
+self.addEventListener('fetch',e=>{});
```
Comment (DiegoRBaquero) [inline static/sw.js:1]: Why is this needed?

**Review by DiegoRBaquero — COMMENTED** [review]
This would be a cool addition, want me to take over?

File: static/manifest.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  "theme_color": "#eee",
+  "description": "Streaming file transfer over WebTorrent (torrents on the web)",
+  "icons": [
+    {
+      "src": "/favicon.ico",
+      "sizes": "64x64",
+      "type": "image/x-icon"
+    }
+  ]
+}
```
Comment (DiegoRBaquero) [inline static/manifest.json:17]:
```suggestion
}

```

File: static/sw.js (hunk 1 above)
Comment (ThaUnknown) [reply static/sw.js:1]: needs a service worker with a fetch event listener to be installable

**Comment by ThaUnknown** [comment]
> This would be a cool addition, want me to take over?

I mean it's as basic as it gets really, you could add more features if you'd like to, but other than that it's pretty much done.

File: static/sw.js (hunk 1 above)
Comment (DiegoRBaquero) [reply static/sw.js:1]:
```suggestion
/* global self */

self.addEventListener('fetch', e => {})

```

**Review by DiegoRBaquero — APPROVED** [review]

**Review by alxhotel — APPROVED** [review]
LGTM
